lib/torrent.js

```diff
@@ -207,6 +207,12 @@ Torrent.prototype._onParsedTorrent = function (parsedTorrent) {
     })
   }
 
+  // destroy DHT client if the torrent is private
+  if (self.parsedTorrent.private && self.client.dht) {
+    self.client.dht.destroy()
+    self.client.dht = null
+  }
```

Comment:
Oops, yeah that doesn't seem right.

Then disabling DHT for the discovery and setting DHT to false when passing the handshake options to bittorrent-protocol should suffice?

I'll take another stab at this.